specs/ClientCertificate.md

```diff
@@ -0,0 +1,500 @@
+                        }
+
+                        // Display custom dialog box for the client certificate selection.
+                        ClientCertificateSelectionDialog dialog(
+                            m_appWindow->GetMainWindow(), L"Select a Certificate for authentication", host.get(), port, client_certificates_);
+
+                        if (dialog.confirmed)
+                        {
+                            int selectedCertificate = dialog.selectedItem;
+                            if (selectedCertificate >= 0)
+                            {
+                                CHECK_FAILURE(
+                                    certificateList->GetValueAtIndex(selectedCertificate, &certificate));
+
+                                CHECK_FAILURE(args->put_SelectedCertificate(certificate));
+                                CHECK_FAILURE(args->put_ClientCertificateRequestResponseState(
+                                    COREWEBVIEW2_CLIENT_CERTIFICATE_REQUEST_RESPONSE_STATE_CONTINUE_WITH_CERTIFICATE));
+                            }
+                            else
+                            {
+                                CHECK_FAILURE(args->put_ClientCertificateRequestResponseState(
+                                    COREWEBVIEW2_CLIENT_CERTIFICATE_REQUEST_RESPONSE_STATE_CONTINUE_WITHOUT_CERTIFICATE));
+                            }
+                        }
+                        else
+                        {
+                            CHECK_FAILURE(args->put_ClientCertificateRequestResponseState(
+                                COREWEBVIEW2_CLIENT_CERTIFICATE_REQUEST_RESPONSE_STATE_CONTINUE_WITHOUT_CERTIFICATE));
+                        }
+                    };
+
+                    // Obtain a deferral for the event so that the CoreWebView2
+                    // doesn't examine the properties we set on the event args and
+                    // after we call the Complete method asynchronously later.
+                    wil::com_ptr<ICoreWebView2Deferral> deferral;
+                    CHECK_FAILURE(args->GetDeferral(&deferral));
+
+                    // This function can be called to show the custom client certificate dialog and
+                    // complete the event at a later time, allowing the developer to
+                    // perform async work before the event completes.
+                    m_completeDeferredCertificateRequestedEvent = [showDialog, deferral] {
+                        showDialog();
+                        CHECK_FAILURE(deferral->Complete());
+                    };
+
+                    return S_OK;
+                })
+                .Get(),
+            &m_ClientCertificateRequestedToken));
+    //! [ClientCertificateRequested2]
+}
+```
+
+## .NET/ WinRT
+```c#
+private bool _isClientCertificateRequested = false;
+void ClientCertificateRequestedCmdExecuted(object target, ExecutedRoutedEventArgs e)
```

Comment:
Looks like this is an ICommand hookup to a enable/disable button in the UI? Difficult to tell without context. I think it would be OK to leave this out.